### PR TITLE
Button should be hidden if the protocol of an endpoint is static

### LINKED ISSUE #26542: Hide the download button for bundles pushed to static environments
### Parent Issue

_No response_

### Problem Statement

Users trying to download bundles pushed to static environments 
This results in 500 error as currently these bundles are not downloadable

Hiding the download button for these bundles is the solution to avoid such scenarios.

### Steps to Reproduce

1. Try downloading bundles pushed to static environments
2. This results in a 500 error 

### Acceptance Criteria

Remove the download button for these bundles 

### dotCMS Version

22.03.10
23.01

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.slack.com/archives/C028Z3R2D/p1698341701309859
https://dotcms.zendesk.com/agent/tickets/113829

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by AndreyDotcms**
We retrieve the selected environments based on the current bundle. Then, we iterate over the obtained list of environments and search for the protocol of every related endpoint. Finally, we validate if the new list contains a static protocol.

image

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp [hunk 1]
```diff
@@ -24,6 +27,15 @@
 
     Bundle bundle = APILocator.getBundleAPI().getBundleById(bundleId);
 
+    List<String> endpointsProtocols = new ArrayList<>();
```
Comment (erickgonzalez) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp:30]: Let's rename this variable to something clearer.

Also let's add some doc

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp [hunk 2]
```diff
@@ -24,6 +27,15 @@
...
+    try {
+        for (Environment e: APILocator.getEnvironmentAPI().findEnvironmentsByBundleId(bundleId)) {
+            pepAPI.findSendingEndPointsByEnvironment(e.getId()).forEach(endpoint -> {
```
Comment (erickgonzalez) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp:33]: Let's rename pepAPI to something easy to understand.

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp (hunk 1 above)
Comment (AndreyDotcms) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp]: Done https://github.com/dotCMS/core/pull/27364/commits/0c4e5e64e4b690bd6461847351c5c7aace6b86cc

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp (hunk 2 above)
Comment (AndreyDotcms) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/publishing/view_publish_audit_detail.jsp]: Done https://github.com/dotCMS/core/pull/27364/commits/0c4e5e64e4b690bd6461847351c5c7aace6b86cc

**Pull request merged**